third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -1742,21 +1742,28 @@ impl Loader {
                 .into_iter()
                 .unzip();
 
-        // For aggregators / snapshots, the first field should be lifted.
-        if let Some(kind) = &maybe_mapping {
-            if let Some(l) = field_layouts.first_mut() {
-                *l = MoveTypeLayout::Tagged(
-                    LayoutTag::IdentifierMapping(kind.clone()),
-                    Box::new(l.clone()),
-                );
-            }
-        }
-
         let has_identifier_mappings =
             maybe_mapping.is_some() || field_has_identifier_mappings.into_iter().any(|b| b);
 
         let field_node_count = *count - count_before;
-        let struct_layout = MoveStructLayout::new(field_layouts);
+        let layout = if Some(IdentifierMappingKind::DerivedString) == maybe_mapping {
```

Comment:
@georgemitenkov @runtian-zhou: take a look at this change, and other changes in this file. 

Special casing for DerivedStrings is a bit different than for other types (whole type is tagged/exchanged, instead of just the value field)